### PR TITLE
CHE-1523: disallow renaming of the classes and packages

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?

Disabled renaming children of the source folder
### What issues does this PR fix or reference?

https://jira.codenvycorp.com/browse/CHE-1523
### Tests written?

Yes

Please review @vparfonov

### CODE REVIEW

**Comment by vparfonov** [comment]
ok

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/resources/RenamingSupport.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+package org.eclipse.che.ide.api.resources;
+
+/**
+ * Indicates resource which allow renaming.
+ *
+ * @author Valeriy Svydenko
+ */
+public interface RenamingSupport {
+    /** Returns {@code} true if renaming is allowed otherwise {@code} false */
+    boolean isRenamed(Resource resource);
```
Comment (vzhukovs) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/resources/RenamingSupport.java]: Maybe `isRenameAllowed()` ?

File: plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/resource/JavaSourceRenameValidator.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Disable rename operation for the children of the source folder.
+ *
+ * @author Valeriy Svydenko
+ */
+public class JavaSourceRenameValidator implements RenamingSupport {
+    @Override
+    public boolean isRenamed(Resource resource) {
+        return !resource.getParentWithMarker(SourceFolderMarker.ID).isPresent();
```
Comment (vzhukovs) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/resource/JavaSourceRenameValidator.java]: Add additional checking for the `.java` extension.

Comment (svor) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/resource/JavaSourceRenameValidator.java]: Why do we need this, I think each element which is in the source folder should be renamed by using refactoring, no?

**Pull request merged**